File: sky/src/types/product.ts
export type ProductTag = {
    tagId: number
    tagName: string
    tagDesc: string
    status: number
    createTime: Date
}
export type Product = {
    productId: number
    productName: string
    categoryId: number
    mainImage: string
    subImages: string
    originalPrice: number
    salePrice: number
    stock: number
    description: string
    status: number
    productTag: ProductTag
    createTime: Date
    updateTime: Date
}
export type ProductItem = {
    categoryId: number
    categoryName: string
    sortOrder: number
    status: number
    products: Product[]
    createTime: Date
}
export type CartProduct = Product & { count: number }

export type toOrder = {
    userId: number
    productId: number
    salePrice: number
    amount: number
    mobile: number
    nickName: string
    address: string
}

File: item-service/src/main/java/com/sky/item/domain/pojo/Product.java
package com.sky.item.domain.pojo;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Product {
    private Integer productId;
    private String  productName;
    private Integer categoryId;
    private String  mainImage;
    private String  subImages;
    private Double  originalPrice;
    private Double  salePrice;
    private Integer stock;
    private String  description;
    private Integer tagId;
    private Integer status;
    private LocalDateTime createTime;
    private LocalDateTime  updateTime;
}
